skimage/exposure/tests/test_histogram_matching.py

```diff
@@ -62,6 +62,7 @@ def test_match_histograms_channel_axis(self, channel_axis):
         reference = np.moveaxis(self.template_rgb, -1, channel_axis)
         matched = exposure.match_histograms(image, reference,
                                             channel_axis=channel_axis)
+        assert matched.dtype == image.dtype
```

Comment:
```suggestion
        assert matched.dtype.kind == image.dtype.kind
```
... to reflect better the update (bug fix) here. The assertion doesn't hold all the time (e.g., if `image` were float16, `matched` would be float32). But maybe it's okay because this one test isn't meant to grasp the overall behaviour?

Reply:
I'm not sure we should really focus on `float16` behavior. Most of the library just breaks with it to my knowledge. Can you provide an other example?

Reply:
Ok. My point wasn't to focus on the float16 case; it's just that `matched.dtype == image.dtype` does hold all the time except when `image` is float16, in which case the assertion relaxes to `matched.dtype.kind == image.dtype.kind`.

Reply:
This particular function involves uint8 images, so this just verifies that those aren't getting changed to float. There is a separate `test_match_histograms_float_dtype` test case below for floating point types that is using `matched.dtype == _supported_float_type(image.dtype)`.

Reply:
@grlee77 perfect, thanks! Please ignore my comment/suggestion.